### DISCUSSION TITLE
What is the Mention's "length" parameter?

### DISCUSSION THREAD

**Discussion by Atomic-Bean** [AsamK/signal-cli#409] [Q&A] [answered]
While working on the JSON output I noticed that "mentions" have a "length" parameter, but I don't really get what it is. The [signal api](https://github.com/signalapp/Signal-Android/blob/9e7c55847e196165940ccb93c7bf55adbd3dac4a/libsignal/service/src/main/java/org/whispersystems/signalservice/api/messages/SignalServiceDataMessage.java#L561) doesn't describe what it does either. So far it always seems to be 1.

**Comment by AsamK** [comment, marked as answer]
I think the index and length parameters define the substring that should be replaced with the mention text (@name).
Currently signal just puts an object replacement character there (￼ U+FFFC) so the length is always 1.
The length field is probably there to future proof the api.

**Reply by Atomic-Bean** [reply]
Yeah, "start" is definitely the index in the string where the mention starts. Thant makes sense, thanks.
